paasta_tools/cli/cmds/status.py

```diff
@@ -1878,6 +1882,250 @@ def print_kafka_status(
     return 0
 
 
+class EtcdLockServerStatus(TypedDict, total=False):
+    observedGeneration: int
+    available: str
+    clientServiceName: str
+
+
+class LockServerStatus(TypedDict, total=False):
+    etcd: EtcdLockServerStatus
+
+
+class VitessClusterCellStatus(TypedDict, total=False):
+    pendingChanges: str
+    gatewayAvailable: str
+
+
+class VitessClusterKeyspaceStatus(TypedDict, total=False):
+    pendingChanges: str
+    cells: List[str]
+    desiredShards: int
+    shards: int
+    readyShards: int
+    updatedShards: int
+    desiredTablets: int
+    tablets: int
+    readyTablets: int
+    updatedTablets: int
+
+
+class VitessDashboardStatus(TypedDict, total=False):
+    available: str
+    serviceName: str
+
+
+class VTAdminStatus(TypedDict, total=False):
+    available: str
+    serviceName: str
+
+
+class OrphanStatus(TypedDict, total=False):
+    reason: str
+    message: str
+
+
+class VitessClusterStatus(TypedDict, total=False):
+    observedGeneration: int
+    globalLockserver: LockServerStatus
+    gatewayServiceName: str
+    vitessDashboard: VitessDashboardStatus
+    cells: Dict[str, VitessClusterCellStatus]
+    keyspaces: Dict[str, VitessClusterKeyspaceStatus]
+    vtadmin: VTAdminStatus
+    orphanedCells: Dict[str, OrphanStatus]
+    orphanedKeyspaces: Dict[str, OrphanStatus]
+
+
+def print_vitess_status(
+    cluster: str,
+    service: str,
+    instance: str,
+    output: List[str],
+    vitess_status: Mapping[str, Any],
+    verbose: int = 0,
+) -> int:
+    tab = "    "
+    indent = 1
+
+    status: VitessClusterStatus = vitess_status.get("status")
+    if status is None:
+        output.append(
+            PaastaColors.red("indent * tab + Vitess cluster is not available yet")
+        )
+        return 1
+
+    output.append(indent * tab + "Vitess Cluster:")
+    indent += 1
+
+    output.append(
+        indent * tab
+        + "Observed Generation: "
+        + str(status.get("observedGeneration", 0))
+    )
+    output.append(
+        indent * tab + "Gateway Service Name: " + status.get("gatewayServiceName", "")
+    )
+
+    output.append(indent * tab + "Cells:")
+    indent += 1
+    cells: Dict[str, VitessClusterCellStatus] = status.get("cells")
+    if not cells:
+        output.append(
+            indent * tab + "Cells: " + PaastaColors.red("No cell status available")
+        )
+        return 0
```

Comment:
just to confirm: for each of these sections, we can't proceed with the subsequent sections if the current one is unavailable?